### PR TITLE
Add ESLint

### PR DESCRIPTION
**Pull request by scripthunter7**
- Install, configure ESLint
- Code refactors

TODO: Is the ESLint config correct?

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (maximtop) [inline .eslintrc]: JS files are preferable for configs. They support comments.

File: rollup.config.js [hunk 2]
```diff
@@ -1,91 +1,91 @@
...
+    input: "./src/index.js",
+    output: [
+        {
+            file: "./dist/ecsstree.cjs",
+            format: "cjs",
+            exports: "auto",
+            sourcemap: false,
+        },
+    ],
+    plugins: [json(), externals(), commonjs({ sourceMap: false }), resolve({ preferBuiltins: false })],
```
Comment (maximtop) [inline rollup.config.js]: This line would be more readable if each plugin were on the new line.

**Review by maximtop — APPROVED** [review]

File: rollup.config.js (hunk 2 above)
Comment (scripthunter7) [reply rollup.config.js]: fixed in https://github.com/scripthunter7/ecsstree/pull/1/commits/116dfe005ceace66585f300b86108f9f71d77c69 , but requires disabling prettier or inserting a comment before the first element

File: .eslintrc (hunk 1 above)
Comment (scripthunter7) [reply .eslintrc]: fixed in https://github.com/scripthunter7/ecsstree/pull/1/commits/9ca0dc0beea024f8d283a2c393d79773faa3c35a

**Review by ameshkov — APPROVED** [review]

File: rollup.config.js (hunk 2 above)
Comment (maximtop) [reply rollup.config.js]: Normally we do not use prettier because eslint covers all cases for lining.

**Review by maximtop — APPROVED** [review]

**Pull request merged**